feat: add employee and accounting page to onboarding flow

nkdengineer opened the pull request:
<!-- If necessary, assign reviewers that know the area or changes well. Feel free to tag any additional reviewers you see fit. -->

### Details
<!-- Explanation of the change or anything fishy that is going on -->

### Fixed Issues
<!---
1. Please postfix `$` with a URL link to the GitHub issue this Pull Request is fixing. For example, `$ https://github.com/Expensify/App/issues/<issueID>`.
2. Please postfix  `PROPOSAL:` with a URL link to your GitHub comment, which contains the approved proposal (i.e. the proposal that was approved by Expensify).  For example, `PROPOSAL: https://github.com/Expensify/App/issues/<issueID>#issuecomment-1369752925`

Do NOT add the special GH keywords like `fixed` etc, we have our own process of managing the flow.
It MUST be an entire link to the github issue and your comment proposal ; otherwise, the linking and its automation will not work as expected.

Make sure this section looks similar to this (you can link multiple issues using the same formatting, just add a new line):

$ https://github.com/Expensify/App/issues/<issueID>
$ https://github.com/Expensify/App/issues/<issueID(comment)>

Do NOT only link the issue number like this: $ #<issueID>
--->
$ https://github.com/Expensify/App/issues/48745
PROPOSAL: https://github.com/Expensify/App/issues/48745#issuecomment-2345255157


### Tests
<!---
Add a numbered list of manual tests you performed that validates your changes work on all platforms, and that there are no regressions present.
Add any additional test steps if test steps are unique to a particular platform.
Manual test steps should be written so that your reviewer can repeat and verify one or more expected outcomes in the development environment.

For example:
1. Click on the text input to bring it into focus
2. Upload an image via copy paste
3. Verify a modal appears displaying a preview of that image
--->
1. Login with a new account
2. In the first onboarding step choose `Manage my team's expenses`
3. Verify that the employee step appears with the correct style as design
4. Select an option and go to the next page
5. Verify that the accounting step appears with the correct style as design
- [x] Verify that no errors appear in the JS console

### Offline tests
<!---
Add any relevant steps that validate your changes work as expected in a variety of network states e.g. "offline", "spotty connection", "slow internet", etc. Manual test steps should be written so that your reviewer and QA testers can repeat and verify one or more expected outcomes. If you are unsure how the behavior should work ask for advice in the `#expensify-open-source` Slack channel.
--->
None
### QA Steps
<!---
Add a numbered list of manual tests that can be performed by our QA engineers on the staging environment to validate that your changes work on all platforms, and that there are no regressions present.
Add any additional QA steps if test steps are unique to a particular platform.
Manual test steps should be written so that the QA engineer can repeat and verify one or more expected outcomes in the staging environment.

For example:
1. Click on the text input to bring it into focus
6. Upload an image via copy paste
7. Verify a modal appears displaying a preview of that image
--->
1. Login with a new account
2. In the first onboarding step choose `Manage my team's expenses`
3. Verify that the employee step appears with the correct style as design
4. Select an option and go to the next page
5. Verify that the accounting step appears with the correct style as design
- [x] Verify that no errors appear in the JS console

### PR Author Checklist
<!--
This is a checklist for PR authors. Please make sure to complete all tasks and check them off once you do, or else your PR will not be merged!
-->

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>


https://github.com/user-attachments/assets/6ee05f90-1d78-43e2-9d57-ac8c984fcc3a


<!-- add screenshots or videos here -->

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/user-attachments/assets/705abe68-f14d-4c30-98e7-4264e05f3454

<!-- add screenshots or videos here -->

</details>

<details>
<summary>iOS: Native</summary>


https://github.com/user-attachments/assets/69af09cf-20e3-40f9-812d-1fd874b5baa2


<!-- add screenshots or videos here -->

</details>

<details>
<summary>iOS: mWeb Safari</summary>


https://github.com/user-attachments/assets/cc2141d8-cc63-4281-af9a-f0022d91c18b


<!-- add screenshots or videos here -->

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/user-attachments/assets/09023d9e-8ee1-41fe-9840-cbd15df0d636

<!-- add screenshots or videos here -->

</details>

<details>
<summary>MacOS: Desktop</summary>


https://github.com/user-attachments/assets/e09ef460-d32b-44b7-9ec9-dc143aebe204


<!-- add screenshots or videos here -->

</details>

nkdengineer committed:
feat: add employee and accounting page to onboarding flow

nkdengineer committed:
fix ts error

nkdengineer committed:
Merge branch 'main' into fix/48745

nkdengineer committed:
add OnboadingListItem

nkdengineer committed:
fix test

nkdengineer committed:
add offline indicator and fix ts bug

shawnborton commented:
@dubielzyk-expensify what are your thoughts on the "Connect your accounting..." text below the headline?
![CleanShot 2024-09-16 at 11 18 56@2x](https://github.com/user-attachments/assets/1228ca91-a528-4bcf-986a-2a2b68c15310)

I am thinking it should be bigger (in our normal font size, 15px) and maybe a bit more margin below it and above the options?

s77rt commented on src/components/SelectionList/OnboardingListItem.tsx:
```diff
@@ -0,0 +1,105 @@
+import BaseListItem from './BaseListItem';
+import type {ListItem, OnboardingListItemProps} from './types';
+
+function OnboardingListItem<TItem extends ListItem>({
```
Can't we just use `RadioListItem`?

nkdengineer replied:
> We use 32px padding on desktop modals and 20px on mobile :)

Because we need the custom padding for desktop and mobile then I created a new component.

s77rt replied:
Using a new component here seems an overkill. The difference is just in the padding value, can you instead pass that as a prop?

nkdengineer replied:
> Using a new component here seems an overkill. The difference is just in the padding value, can you instead pass that as a prop? 

@s77rt 
We don't have any prop to do that now then we need to introduce a new prop. Instead, we can use `RadioListItem` with `leftElement` with `marigin-left` as `12` for desktop and `0` for mobile and add the same `margin-right` for `rightElement`. What do you think?

s77rt replied:
If the row will still take full width (e.g. on hover) then let's do it

nkdengineer replied:
> If the row will still take full width (e.g. on hover) then let's do it

It take full width, updated.

s77rt commented:
The font size is not consist


https://github.com/user-attachments/assets/eba43517-bcf1-4802-b877-7037b0277d44

marcaaron commented on src/pages/OnboardingAccounting/BaseOnboardingAccounting.tsx:
```diff
@@ -0,0 +1,166 @@
+                        return;
+                    }
+
+                    // TODO: call CompleOnboarding API after the back-end PR is complete
```
To clarify, in this comment, I am asking you to [not block on the backend changes](https://github.com/Expensify/App/issues/48745#issuecomment-2347232176). We still must call the API here and send the relevant params:

```
//    [policyID], - Optional policyID that might exist if the user selected to manage their team expenses.
//    [companySize], - The range of employee headcount. Options: "1-10", "11-50", "51-100", "101-1000", "1001+".
//    [userReportedIntegration] - The name of the accounting integration a user told us they use (this is distinct from what they actually have set up).
```

Then we can create a new PR with the task whenever the copy is ready. Let me know if it's clear now! Thanks!

nkdengineer replied:
```
if (!onboardingPolicyID) {
    const {adminsChatReportID, policyID} = Policy.createWorkspace(undefined, true, work);
    Welcome.setOnboardingAdminsChatReportID(adminsChatReportID);
    Welcome.setOnboardingPolicyID(policyID);
} else {
    Policy.updateGeneralSettings(onboardingPolicyID, work);
}
```

@marcaaron In the past we will call this logic to create a new policy in the onboarding work step. For now, we will only generate a policyID then create optimistic policy data in `compleOnboarding` or we will call the same logic above and only pass `policyID` to `compleOnboarding` function.

marcaaron replied:
I didn't entirely understand what you said there. But let's call `CreateWorkspace` it as soon as you select "Manage my team's expenses" then?

nkdengineer replied:
Got it.

nkdengineer replied:
> Since we are not going to use the policy in between steps is there any reason that we create the workspace at that early step? Why not create it just before calling completeOnboarding in the accounting step?

@s77rt The context here.

dubielzyk-expensify commented:
> I am thinking it should be bigger (in our normal font size, 15px) and maybe a bit more margin below it and above the options?

Agree. In the mocks I've specified it as `text` which is our regular body 15px font size. I also have 20px margin between the text and the accounting options so let's use that

dubielzyk-expensify commented:
> The font size is not consistent

Agreed. All headlines should be the same here.

Also the padding here is off:
![CleanShot 2024-09-17 at 13 39 14@2x](https://github.com/user-attachments/assets/3834dc85-8e0c-4323-bdf7-fdefd0413bb0)

The text should all line up vertically on the left like so:
![CleanShot 2024-09-17 at 13 41 13@2x](https://github.com/user-attachments/assets/2abcfc90-74e6-4af4-9b13-09612f1c22d0)

We use 32px padding on desktop modals and 20px on mobile :)

nkdengineer commented:
> Also the padding here is off on @s77rt 's recording:
> The text should all line up vertically on the left like so:

This bug is fixed on the latest code.

nkdengineer committed:
Merge branch 'main' into fix/48745

nkdengineer commented:
> The font size is not consistent

Oh, that is the font-size that we use in the onboarding work step and I used this when creating new pages.

nkdengineer commented:
@dubielzyk-expensify How about the spacing between the title and the description in the accounting step? Is it also 20px?

nkdengineer committed:
fix the font-size issue

nkdengineer commented:
Update two screenshots for desktop and mobile.

https://github.com/user-attachments/assets/62832854-8d2d-41e8-81ff-1208530a0850


https://github.com/user-attachments/assets/09609f62-86e8-403c-82bd-d68d37d5c93a

dubielzyk-expensify commented:
The padding between the headline and the text should be 12px. Like so:
![CleanShot 2024-09-17 at 14 45 59@2x](https://github.com/user-attachments/assets/decd4946-8192-4995-9c8f-fc74410c5ac4)

It looks a bit too big in the screenshots above

nkdengineer committed:
decrease spacing between title and description

nkdengineer commented:
Updated.

https://github.com/user-attachments/assets/3026039c-d2fa-4ba9-bb36-a607f8ebc7ac


https://github.com/user-attachments/assets/a037db9b-e7cc-44a3-8fce-ff562ad42704

s77rt commented:
@nkdengineer There is a missing description in the employees steps

nkdengineer commented:
I see there's no description text for employee step on the final comment.

s77rt commented on src/languages/es.ts:
```diff
@@ -1637,6 +1637,19 @@ export default {
             [CONST.ONBOARDING_CHOICES.CHAT_SPLIT]: 'Chatea y divide gastos con tus amigos',
             [CONST.ONBOARDING_CHOICES.LOOKING_AROUND]: 'Algo más',
         },
+        employees: {
+            title: 'How many empleados do you have?',
+            [CONST.ONBOARDING_COMPANY_SIZE.MICRO]: '1-10 empleados',
+            [CONST.ONBOARDING_COMPANY_SIZE.SMALL]: '11-50 empleados',
+            [CONST.ONBOARDING_COMPANY_SIZE.MEDIUM_SMALL]: '51-100 empleados',
+            [CONST.ONBOARDING_COMPANY_SIZE.MEDIUM]: '101-1,000 empleados',
+            [CONST.ONBOARDING_COMPANY_SIZE.LARGE]: 'More than 1,000 empleados',
+        },
+        accounting: {
+            title: 'Do you use any accounting software?',
+            description: 'Connect your accounting software directly to Expensify',
+            noneOfAbove: 'None of the above',
+        },
```
Copy

nkdengineer replied:
@marcaaron Should we confirm these copies on this PR or will complete on the next PR?

marcaaron replied:
This is already confirmed. We need translations though.

s77rt replied:
@nkdengineer Please ask in Slack for the translations

s77rt commented on src/pages/OnboardingAccounting/BaseOnboardingAccounting.tsx:
```diff
@@ -0,0 +1,167 @@
+                text={translate('common.confirm')}
+                // eslint-disable-next-line rulesdir/prefer-early-return
+                onPress={() => {
+                    if (!userReportedIntegration) {
+                        setError(translate('onboarding.purpose.errorSelection'));
+                        // eslint-disable-next-line no-useless-return
+                        return;
+                    }
```
Check against `undefined` instead of falsy because `null` is a valid value (when selecting none of the above)

s77rt commented on src/pages/OnboardingAccounting/BaseOnboardingAccounting.tsx:
```diff
@@ -0,0 +1,167 @@
+                        return;
+                    }
+
+                    // TODO: call CompleOnboarding API after the back-end PR is complete
```
Call `Welcome.setOnboardingAccounting` before calling the API

nkdengineer replied:
We will call the `compleOnboarding` API here and will pass the account to this function so I think we don't need to do that here.

nkdengineer replied:
I think we can call `enablePolicyConnections` before call `compleOnboarding` API.

s77rt commented:
After refresh and going back I'm taken to the first step instead of the previous step

https://github.com/user-attachments/assets/21440ece-a74f-4757-8cbf-147e851c4bdd

s77rt commented on src/pages/OnboardingAccounting/BaseOnboardingAccounting.tsx:
```diff
@@ -0,0 +1,167 @@
+                </View>
+                <SelectionList
+                    sections={[{data: accountingOptions}]}
+                    onSelectRow={(item) => {
+                        setUserReportedIntegration(item.keyForList);
+                    }}
```
Clear the error on row selection

s77rt commented on src/pages/OnboardingAccounting/index.tsx:
```diff
@@ -0,0 +1,25 @@
+    );
+}
+
+OnboardingAccounting.displayName = 'OnboardingPurpose';
```
```suggestion
OnboardingAccounting.displayName = 'OnboardingAccounting';
```

s77rt commented on src/pages/OnboardingEmployees/index.tsx:
```diff
@@ -0,0 +1,25 @@
+    );
+}
+
+OnboardingEmployees.displayName = 'OnboardingPurpose';
```
```suggestion
OnboardingEmployees.displayName = 'OnboardingEmployees';
```

nkdengineer committed:
Merge branch 'main' into fix/48745

nkdengineer committed:
Update src/pages/OnboardingEmployees/index.tsx

Co-authored-by: Abdelhafidh Belalia <16493223+s77rt@users.noreply.github.com>

nkdengineer committed:
Update src/pages/OnboardingAccounting/index.tsx

Co-authored-by: Abdelhafidh Belalia <16493223+s77rt@users.noreply.github.com>

nkdengineer committed:
call completeOnboarding API

nkdengineer committed:
fix ts error

s77rt commented:
@nkdengineer 

> see there's no description text for employee step on the final comment https://github.com/Expensify/App/issues/48745#issuecomment-2344872759

I think it's missing in the mockup too cc @dubielzyk-expensify

s77rt commented on src/pages/OnboardingAccounting/BaseOnboardingAccounting.tsx:
```diff
@@ -15,23 +17,35 @@ import useResponsiveLayout from '@hooks/useResponsiveLayout';
 import type {OnboardingIcon} from '@src/types/onyx/Onboarding';
 import type {BaseOnboardingAccountingProps} from './types';
 
+type OnboardingAccountingType = ValueOf<typeof CONST.POLICY.CONNECTIONS.NAME>;
```
Move to CONST with the others

s77rt commented on src/pages/OnboardingAccounting/BaseOnboardingAccounting.tsx:
```diff
@@ -112,13 +126,44 @@ function BaseOnboardingAccounting({shouldUseNativeStyles}: BaseOnboardingAccount
                     }
 
-                    // TODO: call CompleOnboarding API after the back-end PR is complete
+                    if (!onboardingPurposeSelected) {
+                        return;
+                    }
```
In which case we reach this step and with no purpose selected?

nkdengineer replied:
We will not reach this case, but it's used to prevent the ts error. The same way is used here.

https://github.com/Expensify/App/blob/8d1972651eefdd89a01cbccc1b0ed0b7240ac9ed/src/pages/OnboardingPersonalDetails/BaseOnboardingPersonalDetails.tsx#L53

s77rt commented on src/pages/OnboardingAccounting/BaseOnboardingAccounting.tsx:
```diff
@@ -112,13 +126,44 @@ function BaseOnboardingAccounting({shouldUseNativeStyles}: BaseOnboardingAccount
+                        return;
+                    }
+
+                    if (onboardingPolicyID) {
+                        Policy.enablePolicyConnections(onboardingPolicyID, true);
+                    }
```
Same ^ This should always be available

marcaaron replied:
We should not call this. It will be done in the backend already.

nkdengineer replied:
@marcaaron Should we update the optimistic data for policy that will enable connection if `userReportedIntegration` is `undefined`?

marcaaron replied:
yes, you've got the idea. If it has a value then we can enable it optimistically. But we don't need  the navigation logic from `enablePolicyConnections()` just the `onyxData` parts.

marcaaron replied:
>enable connection if userReportedIntegration is undefined?

But I think you meant _not_ `undefined`?

nkdengineer replied:
> But I think you meant not undefined?

Yes, I meant that 😄

s77rt replied:
@nkdengineer Shouldn't we remove that line already? (the call to `Policy.enablePolicyConnections`)

nkdengineer replied:
Ah, I missed it.

s77rt commented on src/pages/OnboardingAccounting/BaseOnboardingAccounting.tsx:
```diff
@@ -112,13 +126,44 @@ function BaseOnboardingAccounting({shouldUseNativeStyles}: BaseOnboardingAccount
+                    Report.completeOnboarding(
+                        onboardingPurposeSelected,
+                        CONST.ONBOARDING_MESSAGES[onboardingPurposeSelected],
+                        {
+                            firstName: '',
+                            lastName: '',
+                        },
```
This should be optional.

(To recheck: the BE seems to complain about the empty names)

nkdengineer replied:
Should we also update the default value of this data in `compleOnboarding` function with the default value as empty string?

s77rt replied:
Make it optional in completeOnboarding

marcaaron replied:
AFAICT it is optional. Lmk what you are seeing. Yes I would update this to be optional.

I am also pretty confused about why it's being passed as an object if we can fix that while we are here. Maybe someone thought it better to parameterize the arguments? But the next person disagreed 😄  

Looks inconsistent to me. Let's make it:

```js
function completeOnboarding(
    engagementChoice: OnboardingPurposeType,
    data: ValueOf<typeof CONST.ONBOARDING_MESSAGES>,
    firstName?: string;
    lastName?: string;
    adminsChatReportID?: string,
    onboardingPolicyID?: string,
    paymentSelected?: string,
    ...
) {
```

s77rt replied:
@marcaaron 

```js
{
    "jsonCode": 402,
    "message": "402 Missing firstName",
    "onyxData": [],
    "requestID": "8c542209db944e20-ORN"
}
```

marcaaron replied:
Sigh... lol 😄

nkdengineer replied:
@marcaaron Should we add default value for `firstName` and `lastName` in `completeOnboarding` because in the type `CompleteGuidedSetupParams`, `firstName` and `lastName` is string.

marcaaron replied:
Yes let's do it. We will need it in some cases, but not for the "manage team expense" choice so it is appropriate to make optional.

nkdengineer replied:
I updated.

nkdengineer committed:
Merge branch 'main' into fix/48745

nkdengineer committed:
move the type to CONST

nkdengineer committed:
remove OnboardingListItem

nkdengineer committed:
remove onboardingIcon type

nkdengineer committed:
remove unused style

nkdengineer committed:
fix lint

s77rt commented on src/pages/OnboardingAccounting/BaseOnboardingAccounting.tsx:
```diff
@@ -47,68 +41,68 @@ function BaseOnboardingAccounting({shouldUseNativeStyles, route}: BaseOnboarding
+                    <View style={[styles.mr3, onboardingIsMediumOrLargerScreenWidth ? styles.ml3 : styles.ml0, styles.onboardingIconWrapper]}>
+                        <Icon
+                            src={accountingIcon}
+                            width={40}
+                            height={40}
```
Use `variables.iconSizeExtraLarge`

s77rt commented on src/pages/OnboardingAccounting/BaseOnboardingAccounting.tsx:
```diff
@@ -47,68 +41,68 @@ function BaseOnboardingAccounting({shouldUseNativeStyles, route}: BaseOnboarding
                 text,
-                onboardingIcon,
+                leftElement: (
+                    <View style={[styles.mr3, onboardingIsMediumOrLargerScreenWidth ? styles.ml3 : styles.ml0, styles.onboardingIconWrapper]}>
```
Move the styles to the Icon's `additionalStyles`

s77rt commented on src/pages/OnboardingEmployees/BaseOnboardingEmployees.tsx:
```diff
@@ -30,15 +29,17 @@ function BaseOnboardingEmployees({shouldUseNativeStyles, route}: BaseOnboardingE
     const [selectedCompanySize, setSelectedCompanySize] = useState<OnboardingCompanySizeType | null | undefined>(null);
     const [error, setError] = useState('');
 
-    const companySizeOptions: ListItem[] = useMemo(() => {
```
Why did you remove the type? Can we add it back?

nkdengineer replied:
Added again.

s77rt reviewed (changes requested):
+Conflicts

s77rt commented:
Let's init the employees value from onyx

https://github.com/user-attachments/assets/b51696e7-bef8-49f0-af6b-1d7a6e9c7b0c

nkdengineer committed:
merge main

nkdengineer committed:
add type and refactor style

nkdengineer committed:
Merge branch 'main' into fix/48745

nkdengineer committed:
change firstname and lastname as optional

nkdengineer commented:
> Let's init the employees value from onyx

@s77rt 

```
const [onboardingCompanySize] = useOnyx(ONYXKEYS.ONBOARDING_COMPANY_SIZE);
```
We have a problem in this case: the first time the component is rendered (reloading this page), `onboardingCompanySize` from Onyx is `undefined` because it's stored in cache. That means we can't focus on the selected item by using the `initiallyFocusedOptionKey` prop. 

I added a solution that is to add an `Onyx.connect` in `OnboardingFlow` for this key then the value of the key can be stored in the cache and we can get the correct value the first time the component is rendered.

What do you think? Let's me know if you have another idea.

nkdengineer committed:
migrate to useOnyx

nkdengineer committed:
remove withOnyx

s77rt commented:
@nkdengineer 

> I added a solution that is to add an Onyx.connect in OnboardingFlow for this key then the value of the key can be stored in the cache

This is a workaround. Please remove it.

Also I think the case you mentioned is not worth optimizing for, the case I reported is when we refresh on the accounting page and go back we should have a selection, this should work already without the workaround. The case you mentioned is when we refresh on the employees page, i think it's okay to have no selection.

nkdengineer committed:
Merge branch 'main' into fix/48745

nkdengineer committed:
remove unnecessary code

nkdengineer commented:
@s77rt Updated

> Also I think the case you mentioned is not worth optimizing for, the case I reported is when we refresh on the accounting page and go back we should have a selection, this should work already without the workaround

Yes, it worked.

s77rt commented on src/pages/OnboardingAccounting/types.ts:
```diff
@@ -0,0 +1,12 @@
+import type {OnboardingModalNavigatorParamList} from '@libs/Navigation/types';
+import type SCREENS from '@src/SCREENS';
+
+type OnboardingAccountingProps = Record<string, unknown> & StackScreenProps<OnboardingModalNavigatorParamList, typeof SCREENS.ONBOARDING.ACCOUNTING>;
```
What's is this type `Record<string, unknown>` for?

nkdengineer replied:
removed

s77rt commented on src/pages/OnboardingEmployees/types.ts:
```diff
@@ -0,0 +1,12 @@
+import type {OnboardingModalNavigatorParamList} from '@libs/Navigation/types';
+import type SCREENS from '@src/SCREENS';
+
+type OnboardingEmployeesProps = Record<string, unknown> & StackScreenProps<OnboardingModalNavigatorParamList, typeof SCREENS.ONBOARDING.EMPLOYEES>;
```
Same ^

s77rt commented on src/pages/OnboardingEmployees/BaseOnboardingEmployees.tsx:
```diff
@@ -0,0 +1,109 @@
+                text: translate(`onboarding.employees.${companySize}`),
+                keyForList: companySize,
+                isSelected: companySize === selectedCompanySize,
+                leftElement: onboardingIsMediumOrLargerScreenWidth ? <View style={styles.ml3} /> : null,
+                rightElement: onboardingIsMediumOrLargerScreenWidth ? <View style={styles.mr3} /> : null,
```
Can we apply margin/padding to the main element instead of these fake spacers? Also currently the checkbox goes beyond the intended area

<img width="545" alt="Screenshot 2024-09-20 at 5 24 57 PM" src="https://github.com/user-attachments/assets/a5eddbfa-9aaf-44ff-a525-e2a2e91676a0">

nkdengineer replied:
As I mentioned, `SelectionList` doesn't have a prop to add style to `ListItem`. Then if we want to do that, we need to introduce a new prop.

nkdengineer replied:
@s77rt Updated with a new prop.

s77rt commented on src/pages/OnboardingEmployees/BaseOnboardingEmployees.tsx:
```diff
@@ -0,0 +1,109 @@
+            shouldEnableKeyboardAvoidingView
+            testID="BaseOnboardingEmployees"
+        >
+            <View style={[styles.h100, styles.defaultModalContainer, shouldUseNativeStyles && styles.pt8]}>
```
Remove this outer `View` (not needed) and pass the style to `ScreenWrapper`'s style without `styles.h100`

s77rt commented on src/pages/OnboardingAccounting/BaseOnboardingAccounting.tsx:
```diff
@@ -0,0 +1,216 @@
+            shouldEnableKeyboardAvoidingView
+            testID="BaseOnboardingAccounting"
+        >
+            <View style={[styles.h100, styles.defaultModalContainer, shouldUseNativeStyles && styles.pt8]}>
```
Same ^

s77rt commented on src/CONST.ts:
```diff
@@ -130,12 +130,24 @@ const onboardingEmployerOrSubmitMessage: OnboardingMessageType = {
+    MEDIUM_SMALL: '51-100',
+    MEDIUM: '101-1000',
+    LARGE: '1001+',
+};
```
```suggestion
} as const;
```

s77rt commented on src/pages/OnboardingAccounting/BaseOnboardingAccounting.tsx:
```diff
@@ -0,0 +1,216 @@
+    return (
+        <ScreenWrapper
+            includeSafeAreaPaddingBottom={false}
+            shouldEnableMaxHeight
+            shouldEnableKeyboardAvoidingView
```
Do we need these? We don't have any inputs that require keyboard

nkdengineer replied:
No need

s77rt commented on src/pages/OnboardingEmployees/BaseOnboardingEmployees.tsx:
```diff
@@ -0,0 +1,109 @@
+    return (
+        <ScreenWrapper
+            includeSafeAreaPaddingBottom={false}
+            shouldEnableMaxHeight
+            shouldEnableKeyboardAvoidingView
```
Same ^

s77rt commented on src/pages/OnboardingEmployees/BaseOnboardingEmployees.tsx:
```diff
@@ -0,0 +1,109 @@
+                    progressBarPercentage={onboardingPurposeSelected === CONST.ONBOARDING_CHOICES.MANAGE_TEAM ? 50 : 75}
+                    onBackButtonPress={OnboardingFlow.goBack}
+                />
+                <View style={[onboardingIsMediumOrLargerScreenWidth && styles.mt5, onboardingIsMediumOrLargerScreenWidth ? styles.mh8 : styles.mh5]}>
+                    <View style={[onboardingIsMediumOrLargerScreenWidth ? styles.flexRow : styles.flexColumn, styles.mb5]}>
+                        <Text style={[styles.textHeadlineH1]}>{translate('onboarding.employees.title')}</Text>
+                    </View>
+                </View>
```
Why do we have two Views?

nkdengineer replied:
It's unnecessary, removed

s77rt commented on src/pages/OnboardingAccounting/BaseOnboardingAccounting.tsx:
```diff
@@ -0,0 +1,216 @@
+                success
+                large
+                text={translate('common.confirm')}
+                // eslint-disable-next-line rulesdir/prefer-early-return
```
```suggestion
```

s77rt commented on src/pages/OnboardingAccounting/BaseOnboardingAccounting.tsx:
```diff
@@ -0,0 +1,216 @@
+                onPress={() => {
+                    if (userReportedIntegration === undefined) {
+                        setError(translate('onboarding.purpose.errorSelection'));
+                        // eslint-disable-next-line no-useless-return
```
```suggestion
```

s77rt commented on src/pages/OnboardingEmployees/index.tsx:
```diff
@@ -0,0 +1,25 @@
+import BaseOnboardingEmployees from './BaseOnboardingEmployees';
+import type {OnboardingEmployeesProps} from './types';
+
+function OnboardingEmployees({...rest}: OnboardingEmployeesProps) {
```
```suggestion
function OnboardingEmployees(props: OnboardingEmployeesProps) {
```
No need to destruct props twice. The use of `rest` is wrong here as we are not taking any other props (rest === all)

nkdengineer committed:
Update src/SCREENS.ts

Co-authored-by: Abdelhafidh Belalia <16493223+s77rt@users.noreply.github.com>

nkdengineer committed:
Merge branch 'main' into fix/48745

nkdengineer committed:
Update src/CONST.ts

Co-authored-by: Abdelhafidh Belalia <16493223+s77rt@users.noreply.github.com>

nkdengineer committed:
Update src/pages/OnboardingAccounting/BaseOnboardingAccounting.tsx

Co-authored-by: Abdelhafidh Belalia <16493223+s77rt@users.noreply.github.com>

nkdengineer committed:
refactor code

nkdengineer committed:
Merge branch 'fix/48745' of https://github.com/nkdengineer/App into fix/48745

nkdengineer committed:
fix type error

nkdengineer committed:
remove unnecessary prop

s77rt commented:
@dubielzyk-expensify https://github.com/Expensify/App/pull/49161#issuecomment-2358571095

s77rt commented on src/pages/OnboardingAccounting/BaseOnboardingAccounting.tsx:
```diff
@@ -0,0 +1,208 @@
+                success
+                large
+                text={translate('common.confirm')}
+                // eslint-disable-next-line rulesdir/prefer-early-return
```
```suggestion
```

s77rt reviewed (changes requested):
Almost there

s77rt commented on src/styles/index.ts:
```diff
@@ -4495,6 +4495,13 @@ const styles = (theme: ThemeColors) =>
             backgroundColor: theme.highlightBG,
         },
 
+        onboardingIconWrapper: {
+            alignItems: 'center',
+            justifyContent: 'center',
+            width: 40,
+            height: 40,
+        },
```
Remove this style. It's not needed. We only need `padding: 10` for the `noneAccountingOption`

nkdengineer replied:
@s77rt I updated.

nkdengineer committed:
Merge branch 'main' into fix/48745

nkdengineer committed:
remove unused style

nkdengineer committed:
fix lint

s77rt commented on src/pages/OnboardingAccounting/BaseOnboardingAccounting.tsx:
```diff
@@ -106,7 +105,7 @@ function BaseOnboardingAccounting({shouldUseNativeStyles, route}: BaseOnboarding
                         styles.mr3,
                         onboardingIsMediumOrLargerScreenWidth ? styles.ml3 : styles.ml0,
-                        styles.onboardingIconWrapper,
+                        {padding: 10},
```
Inline styles are discouraged. Let's make a style for this icon

nkdengineer replied:
Updated.

s77rt commented:
@nkdengineer Can you please check https://github.com/Expensify/App/pull/49161#discussion_r1771856090

nkdengineer committed:
add style prop for list item

s77rt commented on src/components/SelectionList/types.ts:
```diff
@@ -263,6 +263,9 @@ type ListItemProps<TItem extends ListItem> = CommonListItemProps<TItem> & {
 
     /** Whether to show RBR */
     shouldDisplayRBR?: boolean;
+
+    /** Styles for the wrapper view */
+    wrapperStyle?: StyleProp<ViewStyle>;
```
Remove. This is already included in `CommonListItemProps`

nkdengineer replied:
Updated.

s77rt commented on src/pages/OnboardingAccounting/BaseOnboardingAccounting.tsx:
```diff
@@ -81,14 +80,9 @@ function BaseOnboardingAccounting({shouldUseNativeStyles, route}: BaseOnboarding
-                            styles.mr3,
-                            onboardingIsMediumOrLargerScreenWidth ? styles.ml3 : styles.ml0,
-                        ]}
+                        additionalStyles={[StyleUtils.getAvatarBorderStyle(CONST.AVATAR_SIZE.DEFAULT, CONST.ICON_TYPE_AVATAR), styles.mr3]}
```
Do we still need `styles.mr3`?

nkdengineer replied:
Yes we still need `mr3` to add the space between the icon and the text.

s77rt commented on src/styles/index.ts:
```diff
@@ -4499,6 +4499,15 @@ const styles = (theme: ThemeColors) =>
+            padding: 10,
+        },
+
+        onboardingWrapper: {
+            paddingLeft: 32,
+            paddingRight: 32,
+        },
```
Remove. Use `styles.ph8`

nkdengineer replied:
I tried to use `styles.ph8` and it doesn't work.

s77rt replied:
Ah I see. It's because `styles.sidebarLinkInner` uses `paddingLeft`/`paddingRight` and those styles has higher priority than `paddingHorizontal`. In that case let's use `[styles.pl8, styles.pr8]`.

Also there is styles.pr9 which is wrong, please rename it to styles.pr8

nkdengineer replied:
Updated.

nkdengineer committed:
Merge branch 'main' into fix/48745

nkdengineer committed:
remove unnecessary prop

nkdengineer committed:
add pl8 and pr8 style

s77rt reviewed (commented):
Looks good overall. I think we are missing:
1. The description in the employees step. Waiting on @dubielzyk-expensify for confirmation https://github.com/Expensify/App/pull/49161#issuecomment-2358571095
2. The translations https://github.com/Expensify/App/pull/49161#discussion_r1771884512

nkdengineer committed:
Merge branch 'main' into fix/48745

s77rt commented:
@nkdengineer Can you resolve conflicts while we wait on the two items ^

nkdengineer committed:
resolve conflict

nkdengineer commented:
@s77rt I updated.

nkdengineer committed:
merge main

nkdengineer committed:
fix ts error

dubielzyk-expensify reviewed (dismissed):
Tried this locally. Looks good from a visual perspective 👍

s77rt commented:
@dubielzyk-expensify Don't we need a description (sub title) in the employees step?

![365538175-65ef58c5-9983-4f77-a138-47839e4ab1d8](https://github.com/user-attachments/assets/7f608085-7748-44e4-a6b7-900d6a670e1e)
![365869589-65dd0fdc-4581-45e6-99ab-2d780a70502a](https://github.com/user-attachments/assets/b8fbf2ec-b3de-4a32-8d4a-c98f84bfd8cb)

dubielzyk-expensify commented:
You can take it out :)

dubielzyk-expensify commented:
Feel free to take out this description text too:

![CleanShot 2024-10-08 at 10 01 14@2x](https://github.com/user-attachments/assets/990befb5-279c-4f80-b63d-cc90076c5bcc)

nkdengineer committed:
merge main

nkdengineer committed:
remove description translation

nkdengineer committed:
Merge branch 'main' into fix/48745

nkdengineer committed:
update translation

nkdengineer commented:
@s77rt I updated the transaction based on [here](https://github.com/Expensify/App/issues/48745#issuecomment-2398805228) and removed the description in accounting page.

s77rt commented on src/pages/OnboardingAccounting/BaseOnboardingAccounting.tsx:
```diff
@@ -0,0 +1,190 @@
+                shouldShowTooltips={false}
+                listItemWrapperStyle={onboardingIsMediumOrLargerScreenWidth ? [styles.pl8, styles.pr8] : []}
+            />
+            {shouldUseNarrowLayout && <OfflineIndicator />}
```
Why is this needed? It's causing double offline indicator on small screens

<img width="468" alt="Screenshot 2024-10-08 at 10 48 20 AM" src="https://github.com/user-attachments/assets/b0c7cd5b-d50e-46ad-b31a-82d98bca6055">

s77rt commented on src/pages/OnboardingAccounting/BaseOnboardingAccounting.tsx:
```diff
@@ -0,0 +1,190 @@
+                    accountingIcon = Expensicons.NetSuiteSquare;
+                    break;
+                }
+                default: {
+                    text = translate('workspace.accounting.intacct');
+                    accountingIcon = Expensicons.IntacctSquare;
+                    break;
+                }
```
Let's not default to any. There is no default. Also this is causing this bug

<img width="468" alt="Screenshot 2024-10-08 at 11 04 06 AM" src="https://github.com/user-attachments/assets/afdb8174-22c2-4a62-bea5-06684aeb10da">

s77rt replied:
Let's also log a warning if we reach the default case

nkdengineer replied:
@s77rt This happens because we added a new connect is `quickbooksDesktop` in the const, I think we only need to not default any, no need to add a log warning.

s77rt commented on src/pages/OnboardingAccounting/BaseOnboardingAccounting.tsx:
```diff
@@ -0,0 +1,190 @@
+                text={translate('common.confirm')}
+                onPress={() => {
+                    if (userReportedIntegration === undefined) {
+                        setError(translate('onboarding.purpose.errorSelection'));
```
Let's move `onboarding.purpose.errorSelection` to `onboarding.errorSelection`

s77rt commented on src/libs/actions/Report.ts:
```diff
@@ -3648,6 +3645,38 @@ function completeOnboarding(
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.POLICY}${onboardingPolicyID}`,
+            value: {
+                areConnectionsEnabled: false,
```
The onboardingPolicy could already have `areConnectionsEnabled` set to `true`. If this API action failed we should revert to the old `policy.areConnectionsEnabled` and not necessary to `false`.

nkdengineer committed:
Merge branch 'main' into fix/48745

nkdengineer committed:
update confirm button to continue button

nkdengineer committed:
fix ts error

nkdengineer committed:
remove unused variable

s77rt commented on src/pages/OnboardingEmployees/BaseOnboardingEmployees.tsx:
```diff
@@ -82,9 +81,7 @@ function BaseOnboardingEmployees({shouldUseNativeStyles, route}: BaseOnboardingE
-                    <Text style={[styles.textHeadlineH1]}>{translate('onboarding.employees.title')}</Text>
-                </View>
+                <Text style={[styles.textHeadlineH1, styles.mb5]}>{translate('onboarding.employees.title')}</Text>
             </View>
```
Do we still need this outer View?

(Same on BaseOnboardingAccounting)

nkdengineer replied:
Removed.

s77rt commented on src/pages/OnboardingPurpose/BaseOnboardingPurpose.tsx:
```diff
@@ -71,7 +73,12 @@ function BaseOnboardingPurpose({shouldUseNativeStyles, shouldEnableMaxHeight, ro
 
                 if (choice === CONST.ONBOARDING_CHOICES.MANAGE_TEAM) {
-                    Navigation.navigate(ROUTES.ONBOARDING_WORK.getRoute(route.params?.backTo));
+                    if (!onboardingPolicyID) {
+                        const {adminsChatReportID, policyID} = Policy.createWorkspace(undefined, true);
+                        Welcome.setOnboardingAdminsChatReportID(adminsChatReportID);
+                        Welcome.setOnboardingPolicyID(policyID);
+                    }
```
Since we are not going to use the policy in between steps is there any reason that we create the workspace at that early step? Why not create it just before calling `completeOnboarding` in the accounting step?

marcaaron replied:
Interesting. Did you have any concern about that or opinion on why it would be better to do one over the other?

s77rt replied:
No strong preference it's just a little questionable that we create the workspace when not really needed yet. Now I think about it more I think we may have a bug:
1. Login as a new user
2. On the onboarding flow, select purpose to manage team <-- This will create a workspace
3. Logout
4. Login in again
5. If you try to complete the onboarding flow you will be creating another workspace since `onboardingPolicyID` is not on onyx

cc @nkdengineer Can you please confirm if you can reproduce this bug?

marcaaron replied:
Oh hmm that's an interesting edge case for sure - I'm not sure how we would completely solve that without designating a specific policy as the "onboarding policy". Though we could potentially check if a user has an existing policy.

s77rt commented on src/pages/OnboardingPurpose/BaseOnboardingPurpose.tsx:
```diff
@@ -71,7 +73,12 @@ function BaseOnboardingPurpose({shouldUseNativeStyles, shouldEnableMaxHeight, ro
                 Welcome.setOnboardingErrorMessage('');
 
                 if (choice === CONST.ONBOARDING_CHOICES.MANAGE_TEAM) {
-                    Navigation.navigate(ROUTES.ONBOARDING_WORK.getRoute(route.params?.backTo));
```
I believe the work step is no longer needed. Let's remove it

nkdengineer committed:
Merge branch 'main' into fix/48745

nkdengineer committed:
remove onboaring work

nkdengineer committed:
merge main

nkdengineer committed:
remove import type

s77rt commented:
## Reviewer Checklist

- [X] I have verified the author checklist is complete (all boxes are checked off).
- [X] I verified the correct issue is linked in the `### Fixed Issues` section above
- [X] I verified testing steps are clear and they cover the changes made in this PR
    - [X] I verified the steps for local testing are in the `Tests` section
    - [X] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [X] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [X] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [X] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [X] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [X] I verified tests pass on **all platforms** & I tested again on:
    - [X] Android: Native
    - [X] Android: mWeb Chrome
    - [X] iOS: Native
    - [X] iOS: mWeb Safari
    - [X] MacOS: Chrome / Safari
    - [X] MacOS: Desktop
- [X] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [X] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [X] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [X] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [X] I verified that comments were added to code that is not self explanatory
    - [X] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [X] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [X] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [X] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [X] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [X] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [X] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [X] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [X] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [X] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [X] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [X] If a new component is created I verified that:
    - [X] A similar component doesn't exist in the codebase
    - [X] All props are defined accurately and each prop has a `/** comment above it */`
    - [X] The file is named correctly
    - [X] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [X] The only data being stored in the state is data necessary for rendering and nothing else
    - [X] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [X] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [X] All JSX used for rendering exists in the render method
    - [X] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [X] If any new file was added I verified that:
    - [X] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [X] If a new CSS style is added I verified that:
    - [X] A similar style doesn't already exist
    - [X] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [X] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [X] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [X] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [X] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [X] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [X] I verified that all the inputs inside a form are aligned with each other.
    - [X] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [X] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [X] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [X] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

<!-- add screenshots or videos here -->
<img width="480" alt="android" src="https://github.com/user-attachments/assets/0bb9cfa1-8f35-4a56-b65b-cdfba47d8dc7">

</details>

<details>
<summary>Android: mWeb Chrome</summary>

<!-- add screenshots or videos here -->
<img width="480" alt="mweb-chrome" src="https://github.com/user-attachments/assets/fb4fe547-bd4c-41ba-ac6a-dc8de2857bd0">

</details>

<details>
<summary>iOS: Native</summary>

<!-- add screenshots or videos here -->
<img width="559" alt="ios" src="https://github.com/user-attachments/assets/4f79ecfd-bd02-4e30-9ee9-bd09e6ae904e">

</details>

<details>
<summary>iOS: mWeb Safari</summary>

<!-- add screenshots or videos here -->
<img width="559" alt="mweb-safari" src="https://github.com/user-attachments/assets/adb2cb1b-fdc0-4287-bd09-64c7c3b9c3c8">

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

<!-- add screenshots or videos here -->
<img width="1728" alt="web" src="https://github.com/user-attachments/assets/48c1dab1-9daa-4e3b-b53a-746425ac5738">

</details>

<details>
<summary>MacOS: Desktop</summary>

<!-- add screenshots or videos here -->
<img width="1312" alt="desktop" src="https://github.com/user-attachments/assets/41dac0bc-6bb8-44ea-b1e4-960a4048571b">

</details>

marcaaron commented:
Sorry, @nkdengineer. Changes look great, but we have conflicts now.

nkdengineer committed:
merge main

nkdengineer commented:
@marcaaron I updated.

marcaaron commented:
Conflicts again.

nkdengineer committed:
merge main

nkdengineer commented:
@marcaaron I resolved.

marcaaron reviewed (approved):
Looking great! Thanks for your patience and hard work here. And great review as always @s77rt 🙇

OSBotify commented:
:hand: This PR was not deployed to staging yet because QA is ongoing. It will be automatically deployed to staging after the next production release.

marcaaron commented:
@nkdengineer are you working on [adding the task next](https://github.com/Expensify/App/issues/48745#issuecomment-2398812091)? Just making sure you are aware of that part of this ticket. Please let me know.

marcaaron commented:
Heads up that there is another bug that we found concerning the signup flow between OldDot and NewDot. Because we have a flow where we set the `signupQualifier` before the user arrives at this flow. We will then skip the step that creates the policy.

I'm going to be working on that one because it essentially blocks the deploy since we can't release a version of the app where this policy doesn't get created.

s77rt commented on src/pages/OnboardingAccounting/BaseOnboardingAccounting.tsx:
```diff
@@ -0,0 +1,192 @@
+                        userReportedIntegration,
+                    );
+
+                    Welcome.setOnboardingAdminsChatReportID();
+                    Welcome.setOnboardingPolicyID();
```
@nkdengineer Maybe I missed that but do you recall why we clear the onboarding admin report id and policy id?

s77rt replied:
Ah yeah that makes sense! I will look into older PRs then. Thanks!
